activesupport/lib/active_support/inflector/methods.rb

```diff
@@ -195,12 +197,7 @@ def dasherize(underscored_word)
     #
     # See also #deconstantize.
     def demodulize(path)
-      path = path.to_s
```

Comment:
It does make sense for non-string-like objects, but it also breaks
compatibility. Is anyone calling this with symbols, for example?
On Mon, May 2, 2016 at 08:00 Alfonso Jiménez <notifications@github.com>
wrote:

> In activesupport/lib/active_support/inflector/methods.rb
> <https://github.com/rails/rails/pull/23985#discussion_r61750121>:
>
> > @@ -195,12 +197,7 @@ def dasherize(underscored_word)
> >      #
> >      # See also #deconstantize.
> >      def demodulize(path)
> > -      path = path.to_s
>
> @jeremy <https://github.com/jeremy> Thanks for your comments. I've
> rebased the branch.
>
> Must this method be responsible for the string coercion? After these
> changes this will not work anymore:
>
> ActiveSupport::Inflector.demodulize(:'Foo::Bar')
>
> or
>
> ActiveSupport::Inflector.demodulize(BigDecimal.new('1.00'))
>
> But I can't think of a use case for these examples. IHMO, it makes more
> sense to leave that responsibility outside this method.
>
> cc @rafaelfranca <https://github.com/rafaelfranca>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/rails/rails/pull/23985/files/0fbc14189814e1e48dc48c391e7a3191316f5839#r61750121>
>